include/libdnf5/repo/repo.hpp

```diff
@@ -385,6 +372,21 @@ class Repo {
     std::unique_ptr<SolvRepo> solv_repo;
 
     WeakPtrGuard<Repo, false> data_guard;
+
+    bool loaded{false};
+
+    /// Downloads repository metadata.
+    // @replaces libdnf:repo/Repo.hpp:method:Repo.downloadMetadata(const std::string & destdir)
+    void download_metadata(const std::string & destdir);
+
+    /// Loads the repository objects into sacks.
+    ///
+    /// Also writes the libsolv's solv/solvx cache files.
+    void load();
+
+    /// Append a rpm database into the system repository. The type of the repo must be Type::SYSTEM.
+    // TODO(jrohel) this will add packages with conflicting rpmdb ids, which will break some operations
+    void load_extra_system_repo(const std::string & rootdir);
```

Comment:
Please rebase your branch against current `main` state. These three methods: `download_metadata()`, `load()` and `load_extra_system_repo()` are already marked as deprecated or not API, so we don't have to do anything there. As we have already moved to stable API we cannot easily just drop public methods.